app/main/views/questionnaire.py

```diff
@@ -62,9 +62,16 @@ def get_introduction(eq_id, form_type, collection_id):
 @questionnaire_blueprint.route('<location>', methods=["GET"])
 @login_required
 def get_questionnaire(eq_id, form_type, collection_id, location):
-    get_questionnaire_manager(g.schema, g.schema_json).build_state(location, get_answers(current_user))
+    questionnaire_manager = get_questionnaire_manager(g.schema, g.schema_json)
 
-    return _render_template(location, get_questionnaire_manager(g.schema, g.schema_json).state, template='questionnaire')
+    questionnaire_manager.build_state(location, get_answers(current_user))
+
+    schema = questionnaire_manager.get_schema()
+    block = schema.get_item_by_id(location) if schema.item_exists(location) else None
```

Comment:
it won't call schema.get_item_by_id(location) if it doesn't exist as it checks if schema.item_exists(location)